sdks/java/core/src/test/java/org/apache/beam/sdk/runners/inprocess/InMemoryWatermarkManagerTest.java

```diff
@@ -841,6 +852,46 @@ public void synchronizedProcessingInputTimeIsHeldToPendingBundleTimes() {
 
     clock.set(BoundedWindow.TIMESTAMP_MAX_VALUE);
     assertThat(downstreamWms.getSynchronizedProcessingInputTime(), equalTo(upstreamHold));
+
```

Comment:
I think this probably should have been in the test in the first place, although maybe slightly changed - we demonstrated before that we hold the SynchronizedProcessingTimeInputWatermark when there are pending elements, but should also verify that when we complete those elements we free the hold.